compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4782,6 +4783,13 @@ private STNode parseObjectMemberWithoutMeta(STNode metadata, ParserRuleContext c
                 STNode semicolonToken = parseSemicolon();
                 member = STNodeFactory.createTypeReferenceNode(asterisk, type, semicolonToken);
                 break;
+            case FINAL_KEYWORD:
```

Comment:
Shall we treat final as a qualifier (similar to "private"/"public")? I think that will simplify many things.